### PR TITLE
Introduce categories to ConfigEntry and migrate configurations

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

This PR introduces `categories` to `ConfigEntry`, we can benefit from it in generating docs.

And this PR migrates a bunch of configurations occur on README to the `ConfigEntry`, note that all of the migrations are compatible w/ early versions.

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: conf/celeborn-defaults.conf.template [hunk 1]
```diff
@@ -24,19 +24,26 @@ rss.rpc.io.numConnectionsPerPeer                 8
...
+celeborn.ha.master.node.1.host                   clb-1
+celeborn.ha.master.node.1.port                   9097
+celeborn.ha.master.node.1.ratis.port             9872
+celeborn.ha.master.node.2.host                   clb-2
+celeborn.ha.master.node.2.port                   9098
+celeborn.ha.master.node.2.ratis.port             9873
+celeborn.ha.master.node.3.host                   clb-3
+celeborn.ha.master.node.3.port                   9099
+celeborn.ha.master.node.3.ratis.port             9874
+celeborn.ha.master.ratis.raft.server.storage.dir                           /mnt/disk1/rss_ratis/
```
Comment (pan3793) [inline conf/celeborn-defaults.conf.template:47]: all ratis configurations respect the ratis raw properties, w/ additional prefix `celeborn.ha.master.ratis.`

File: common/src/main/scala/org/apache/celeborn/common/RssConf.scala [hunk 2]
```diff
@@ -713,17 +903,92 @@ object RssConf extends Logging {
...
+  def haMasterRatisSnapshotAutoTriggerEnabled(conf: RssConf): Boolean =
+    conf.get(HA_MASTER_RATIS_SNAPSHOT_AUTO_TRIGGER_ENABLED)
+
+  def haMasterRatisSnapshotAutoTriggerThreshold(conf: RssConf): Long =
+    conf.get(HA_MASTER_RATIS_SNAPSHOT_AUTO_TRIGGER_THRESHOLD)
+  def haMasterRatisSnapshotRetentionFileNum(conf: RssConf): Int =
+    conf.get(HA_MASTER_RATIS_SNAPSHOT_RETENTION_FILE_NUM)
+
+  val WORKER_REPLICATE_THREADS: ConfigEntry[Int] =
+    buildConf("celeborn.worker.replicate.threads")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/RssConf.scala:951]: I have searched the spark codebase, it has two styles to represent thread numbers, "threads" and "numThreads", I think we should unify the representation in celeborn, personally, I lean toward the shorter.

File: common/src/main/scala/org/apache/celeborn/common/RssConf.scala [hunk 3]
```diff
@@ -713,17 +903,92 @@ object RssConf extends Logging {
...
+        "directory. For the sake of performance, there should be no more than 2 directories " +
+        "on the same disk partition if you are using HDD. There can be 4 or more directories " +
+        "can run on the same disk partition if you are using SSD. For example: " +
+        "dir1[:capacity=][:disktype=][:flushthread=],dir2[:capacity=][:disktype=][:flushthread=]")
+      .stringConf
+      .toSequence
+      .createOptional
+
+  val WORKER_FLUSH_BUFFER_SIZE: ConfigEntry[Long] =
+    buildConf("celeborn.worker.flush.buffer.size")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/RssConf.scala:980]: I see other places using `rss.fluster.*`, maybe we should unify them into `celeborn.worker.flush.*` later

File: common/src/main/scala/org/apache/celeborn/common/RssConf.scala (hunk 2 above)
Comment (waitinfuture) [reply common/src/main/scala/org/apache/celeborn/common/RssConf.scala:972]: +1

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**